src/table/boot.rs

```diff
@@ -390,6 +390,10 @@ pub enum Tpl {
     ///
     /// Even processor interrupts are disable at this level.
     HighLevel = 31,
+    // SAFETY: The UEFI specification repeatedly states that only the these
+    //         priority levels may be used, the rest being reserved for internal
+    //         firmware use. So only these priority levels should be exposed to
+    //         the application, and modeling them as a Rust enum seems safe.
```

Comment:
I am not 100% sure about this one. I have not studied the specification well enough to tell, but in principle one could accidentally get exposed to the TPL of internal firmware tasks with some interfaces. If so, this enum should be newtype-d too.

Reply:
More generally, the safety of enums should in principle be reevaluated anytime a new interface that exports enum values from the firmware is introduced. Not sure where/if I should document this.

Reply:
Yeah, raising the TPL will return the old TPL, which in some very weird corner cases could return the old (firmware specific) TPL. Although Rust code shouldn't even mess with this returned TPL (it should only handle it as an opaque value and pass it to the restore TPL function).

Reply:
True, user code shouldn't mess with it, but since the mere act of writing it down to a variable is UB, I prefer changing this one too :crab: